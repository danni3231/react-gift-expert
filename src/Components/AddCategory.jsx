import { useState } from "react";
import { Input } from "../Styles/Index";

export const AddCategory = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState("");

	const onInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	const onSubmit = (e) => {
		e.preventDefault();
		const newInputValue = inputValue.trim();
		if (newInputValue.length <= 1) return;

		onNewCategory(newInputValue);
		setInputValue("");
	};

	return (
		<form onSubmit={onSubmit}>
			<Input
				type='text'
				placeholder='Buscar Gifs'
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	);
};
