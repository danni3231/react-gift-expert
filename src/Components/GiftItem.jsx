import PropTypes from 'prop-types'
import { Card } from "../Styles/Index";

export const GiftItem = ({ url, title }) => {
	return (
		<Card>
			<img src={url} alt={title} />
			<p>{title}</p>
		</Card>
	);
};

GiftItem.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
};