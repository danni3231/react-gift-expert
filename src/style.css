* {
	font-family: Helvetica, Arial, sans-serif;
	background-color: rgb(234, 234, 234);
}

body {
	padding: 60px;
}

h2 {
	font-size: 1.5rem;
}

h3 {
	font-size: 3rem;
	margin-bottom: 5px;
}
