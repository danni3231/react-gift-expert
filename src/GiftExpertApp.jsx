import { useState } from "react";
import { AddCategory, GifGrid } from "./Components";

export const GiftExpertApp = () => {
	const [categories, setCategories] = useState(["One Punch"]);

	const addCategory = (newCategory) => {
		if (categories.includes(newCategory)) return;
		setCategories([newCategory, ...categories]);
	};

	console.log(categories);

	return (
		<>
			<h1>GiftExpertApp</h1>

			<AddCategory onNewCategory={addCategory} />

			{categories.map((category) => {
				return <GifGrid key={category} category={category} />;
			})}
		</>
	);
};
