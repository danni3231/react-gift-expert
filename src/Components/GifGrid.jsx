import { GiftItem } from "./GiftItem";
import { useFetchGifs } from '../Hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

	const {images, isLoading} = useFetchGifs(category);

	return (
		<>
			<h3>{category}</h3>

			{isLoading && <h2>Loading...</h2>}

			<section className="card-grid">
				{images.map((img) => (
					<GiftItem key={img.id} {...img} />
				))}
			</section>
		</>
	);
};
