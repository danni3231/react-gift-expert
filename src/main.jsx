import React from "react";
import ReactDom from "react-dom/client";
import { GiftExpertApp } from "./GiftExpertApp";

import "./style.css";

ReactDom.createRoot(document.getElementById("root")).render(
	<React.StrictMode>
		<GiftExpertApp />
	</React.StrictMode>
);
