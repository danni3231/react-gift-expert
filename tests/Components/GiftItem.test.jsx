import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/Components";

describe("Test in <Giftitem/>", () => {
	const { id, title, url } = {
		id: "ABC",
		title: "Un titulo",
		url: "https://localhost/cualquier/cosa.jpg",
	};

	test("debe hacer match con el snapshot", () => {
		const { container } = render(<GiftItem title={title} url={url} />);
		expect(container).toMatchSnapshot();
	});

	test("debe mostrar la imagen con el url y el alt indicado", () => {
		render(<GiftItem title={title} url={url} />);
		//screen.debug();
		const { src, alt } = screen.getByRole("img");
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test("debe mostrar el titulo del componente", () => {
        render(<GiftItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});
