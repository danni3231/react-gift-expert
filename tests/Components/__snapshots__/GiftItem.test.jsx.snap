// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Giftitem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="sc-gjcoXW gVMyRK"
  >
    <img
      alt="Un titulo"
      src="https://localhost/cualquier/cosa.jpg"
    />
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
